middle_end/flambda2/simplify/simplify_apply_expr.ml

```diff
@@ -318,8 +321,9 @@ let simplify_direct_full_application ~simplify_expr dacc apply function_type
               let denv =
                 List.fold_left2
                   (fun denv kind result ->
+                    let result_var, result_uid = BP.var_and_uid result in
                     DE.add_variable denv
-                      (VB.create (BP.var result) NM.in_types)
+                      (VB.create result_var result_uid NM.in_types)
```

Comment:
Since this is in-types mode, the uid could be dropped, but maybe better to propagate since it's right here